### PR TITLE
Fix fixes

### PR DESCRIPTION
**Pull request by Camper12345**
* Fixed mob deafness in closets
* Fixed catwalk footstep sounds shit

### CODE REVIEW

File: code/game/atoms.dm [hunk 1]
```diff
@@ -530,3 +530,6 @@ its easier to just keep the beam vertical.
 /turf/Entered(var/atom/movable/AM, var/atom/old_loc, var/special_event)
 	return ..(AM, old_loc, 0)
 
+/atom/proc/get_footstep_sound()
```
Comment (LethalGhost) [inline code/game/atoms.dm]: Make `get_footstep_sound()` turf proc?

File: code/game/turfs/turf.dm [hunk 2]
```diff
@@ -231,3 +231,14 @@ var/const/enterloopsanity = 100
 
 /turf/proc/update_blood_overlays()
 	return
+
+/turf/get_footstep_sound(var/mobtype)
+	if(!footstep_sounds || !footstep_sounds.len)
+		return
+
+	var/sound = pick(footstep_sounds[mobtype])
```
Comment (LethalGhost) [inline code/game/turfs/turf.dm]: May be replace with `safepick()`

File: code/game/turfs/turf.dm [hunk 3]
```diff
@@ -231,3 +231,14 @@ var/const/enterloopsanity = 100
...
+
+	if(locate(/obj/structure/catwalk) in src)
```
Comment (LethalGhost) [inline code/game/turfs/turf.dm]: Can rewrite `pick` in 239. Better check that first?

File: code/modules/mob/living/carbon/human/human_movement.dm [hunk 4]
```diff
@@ -110,42 +110,41 @@
...
+
+			if(!has_gravity(src))
+				if(step_count % 3) //this basically says, every three moves make a noise
+					return //1st - none, 1%3==1, 2nd - none, 2%3==2, 3rd - noise, 3%3==0
+
+			if(species.silent_steps)
+				return //species is silent
+
+			playsound(T, S, volume, 1, range)
+			return
```
Comment (LethalGhost) [inline code/modules/mob/living/carbon/human/human_movement.dm]: Hm. Better move checks like that (can stop proc): 
```
if(species.silent_steps)
	return //species is silent
```
upper than setting up sound.

File: code/modules/mob/living/say.dm [hunk 5]
```diff
@@ -242,13 +242,10 @@ proc/get_radio_key_from_channel(var/channel)
...
 		var/list/hear = hear(message_range,T)
-		var/list/hearturfs = list()
 
-		for(var/I in hear)
-			if(ismob(I))
-				var/mob/M = I
-				listening += M
-				hearturfs += M.locs[1]
+		for(var/mob/M in mob_list)
+			if(M && M.locs.len && M.locs[1] in hear)
```
Comment (LethalGhost) [inline code/modules/mob/living/say.dm]: `M` can not be `null`. 
`for(var/TYPE/VAR in LIST)`  implicitly call `istype(VAR, TYPE)`

File: code/game/atoms.dm (hunk 1 above)
Comment (Camper12345) [inline code/game/atoms.dm]: footstep_sounds list defined in atom, so this proc too.

Comment (LethalGhost) [inline code/game/atoms.dm]: Move both?

**Pull request merged**